libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/OAuth2.java

```diff
@@ -389,7 +389,12 @@ public static final IdServiceResponse callIdentityService(HttpAccess httpAccesso
      * @param authToken Access token.
      */
     public static final Request.Builder addAuthorizationHeader(Request.Builder builder, String authToken) {
-        return builder.header(AUTHORIZATION, BEARER + authToken);
+        if (builder.getHeaders$okhttp().get(AUTHORIZATION) == null) {
+            return builder.header(AUTHORIZATION, BEARER + authToken);
+        } else {
+            // Don't override header for Native Login authentication.
+            return builder;
+        }
```

Comment:
This feels a bit dirty if you consider that we probably just shouldn't add the interceptor that inserts this authorization header to unauthenticated requests, but attempting that gets pretty complicated.  The client builder is associated with the current account, not the rest client being used.  So even if we are using an unauthenticated rest client, the rest client of the current user is till attached.  

The only scenario in which this matter is Biometric Auth with Native Login.